### PR TITLE
Implement `krun_add_vsock_port()` and `UnixProxy` for guest communication with host UNIX sockets.

### PR DESCRIPTION
**Pull request by tylerfanelli**
This PR introduces two main components:

1) `krun_add_vsock_port()`, which allows a user to map guest vsock port numbers to UNIX IPC sockets running on the host. With this, users can enable VM guest IPC with UNIX sockets running on a host.

2) `vsock::unix` module, implementing the `UnixProxy`, a vsock proxy that facilitates the communication between a guest vsock and UNIX IPC socket running on the host.

### CODE REVIEW

File: src/devices/src/virtio/vsock/muxer.rs [hunk 1]
```diff
@@ -518,10 +518,11 @@ impl VsockMuxer {
                     addr: Ipv4Addr::new(0, 0, 0, 0),
                     port: 0,
                 };
-                unix.connect(pkt, tsi);
+                let update = unix.connect(pkt, tsi);
                 unix.confirm_connect(pkt);
                 proxy_map.insert(id, Mutex::new(Box::new(unix)));
                 ipc_map.remove(&pkt.dst_port());
```
Comment (slp) [inline src/devices/src/virtio/vsock/muxer.rs:524]: What's the reason to remove the port from the map? That means the connection can only the established once, doesn't it?

Comment (tylerfanelli) [reply src/devices/src/virtio/vsock/muxer.rs:524]: Right. Do we want to keep it around for multiple connections?

Comment (slp) [reply src/devices/src/virtio/vsock/muxer.rs:524]: I think I would expect that to be the case. Sounds safer for now.

Comment (tylerfanelli) [reply src/devices/src/virtio/vsock/muxer.rs]: Fixed.

**Review by slp — APPROVED** [review]
LGTM, thanks!

**Pull request merged**